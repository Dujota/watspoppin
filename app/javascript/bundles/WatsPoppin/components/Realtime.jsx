import React from 'react';
import PropTypes from 'prop-types';

import Linkify from 'react-linkify';

export default class Realtime extends React.Component {
  static propTypes = {
    stories: PropTypes.array.isRequired,
    trending_hashtags: PropTypes.array.isRequired,
    search: PropTypes.func.isRequired
  };

  constructor(props) {
    super(props);

    this.state = {
      collapsed: true
    };
  }

  triggerSearch = (e, trend) => {
    e.preventDefault();
    e.stopPropagation();
    this.props.search(trend);
  };

  render() {
    return (
      <div>
        <div className="stories">
          {this.props.stories.map(story => (
            <div key={story.id} className="realtime_story">
              <img
                className={story.image.thumb.url ? 'user_img' : 'default_img'}
                src={
                  story.image.thumb.url
                    ? story.image.thumb.url
                    : '/twitter-logo.svg'
                }
                alt={story.content}
              />
              <p className="handle">{story.handle}</p>
              <Linkify>
                <p>{story.content.substring(0, 600)}...</p>
              </Linkify>
            </div>
          ))}
        </div>

        <div
          className={`trends ${this.state.collapsed ? 'collapsed' : ''}`}
          onClick={e => {
            e.preventDefault();
            this.setState({
              collapsed: !this.state.collapsed
            });
          }}
        >
          <p>TRENDING</p>

          <section className="trending_hashtags">
            <ul>
              {this.props.trending_hashtags.map(trend => (
                <li key={trend}>
                  <a
                    className="trend_link"
                    href="#"
                    onClick={e => this.triggerSearch(e, trend)}
                  >
                    {trend}
                  </a>
                </li>
              ))}
            </ul>
          </section>
        </div>
      </div>
    );
  }
}
