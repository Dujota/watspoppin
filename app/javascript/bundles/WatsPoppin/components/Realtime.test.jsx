import Realtime from './Realtime';
import { spy } from 'sinon';

const stories = [
  {
    id: 30,
    image: { thumb: { url: '/mario.jpg' } },
    content:
      'Locavore mumblecore XOXO seitan, jean shorts williamsburg flannel pickled',
    handle: 'Gorvachov puff puff'
  }
];

const trending_hashtags = ['dog', 'cat', 'bunny'];

describe('Realtime', () => {
  it('Renders correctly', () => {
    const search = spy();

    const wrapper = render(
      <Realtime
        stories={stories}
        trending_hashtags={trending_hashtags}
        search={search}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('Calls search when a trend is clicked', () => {
    const search = spy();

    const wrapper = mount(
      <Realtime
        stories={stories}
        trending_hashtags={trending_hashtags}
        search={search}
      />
    );

    wrapper
      .find('.trend_link')
      .first()
      .simulate('click');

    expect(search.called).toBe(true);
  });

  it('Changes state#collapsed on click', () => {
    const wrapper = shallow(
      <Realtime
        stories={stories}
        trending_hashtags={trending_hashtags}
        search={() => {}}
      />
    );

    expect(wrapper.state().collapsed).toBe(true);

    wrapper
      .find('.trends')
      .first()
      .simulate('click', { preventDefault: () => {} });

    expect(wrapper.state().collapsed).toBe(false);
  });
});
