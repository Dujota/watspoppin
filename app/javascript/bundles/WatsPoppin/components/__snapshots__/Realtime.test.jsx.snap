// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Realtime Renders correctly 1`] = `
<div>
  <div
    class="stories"
  >
    <div
      class="realtime_story"
    >
      <img
        alt="Locavore mumblecore XOXO seitan, jean shorts williamsburg flannel pickled"
        class="user_img"
        src="/mario.jpg"
      />
      <p
        class="handle"
      >
        Gorvachov puff puff
      </p>
      <p>
        Locavore mumblecore XOXO seitan, jean shorts williamsburg flannel pickled...
      </p>
    </div>
  </div>
  <div
    class="trends collapsed"
  >
    <p>
      TRENDING
    </p>
    <section
      class="trending_hashtags"
    >
      <ul>
        <li>
          <a
            class="trend_link"
            href="#"
          >
            dog
          </a>
        </li>
        <li>
          <a
            class="trend_link"
            href="#"
          >
            cat
          </a>
        </li>
        <li>
          <a
            class="trend_link"
            href="#"
          >
            bunny
          </a>
        </li>
      </ul>
    </section>
  </div>
</div>
`;
